pkgs/applications/networking/p2p/pyrosimple/default.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, python3Packages
+, nix-update-script
+, pyrosimple
+, testers
+}:
+
+python3Packages.buildPythonPackage rec {
+  pname = "pyrosimple";
+  version = "2.7.0";
+
+  src = python3Packages.fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-SMqzvTbWFHwnbMQ+6K0m1v+PybceQK5EHEuN8FB6SaU=";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    bencode-py
+    APScheduler
+    jinja2
+    python-daemon
+    importlib-resources
+    parsimonious
+    prometheus-client
+    prompt-toolkit
+    requests
+    shtab
+    inotify
+    python-box
+    tomli
+    tomli-w
+  ];
+
+  passthru = {
+    updateScript = nix-update-script { };
+    tests = testers.testVersion { package = pyrosimple; command = "pyroadmin --version"; };
+  };
+
+  meta = with lib; {
+    homepage = "https://kannibalox.github.io/pyrosimple/";
+    description = "A rTorrent client and Python 3 fork of the pyrocore tools";
+    license = licenses.gpl3Plus;
+    changelog = "https://github.com/kannibalox/pyrosimple/blob/v${version}/CHANGELOG.md";
+    platforms   = platforms.all;
```

Comment:
It'll work without `inotify` on other platforms.
It's listed [upstream as an optional dependency](https://github.com/kannibalox/pyrosimple/blob/6fee78e6342890c6503935a73f698abd4c6f5441/pyproject.toml#L43).

The functionality dependent on file system monitoring doesn't work, but otherwise it works fine.

I _think_ moving the dependencies to an argument, and having `inotify` omitted if the platform is not linux would be the right way to do this.